### PR TITLE
machines: quering of graphics support should use virsh domcapabilities

### PR DESCRIPTION
**Pull request by KKoukiou**
Previously only spice support was checked, now both spice and vnc support are
quered with virsh domcapabilities.

- [x] ack from @Antique

### CODE REVIEW

File: pkg/machines/scripts/create_machine.sh [hunk 1]
```diff
@@ -41,10 +26,17 @@ else
     DISK_OPTIONS="size=$STORAGE_SIZE,format=qcow2"
 fi
 
-GRAPHICS_PARAM="--graphics vnc,listen=127.0.0.1"
-if spiceSupported; then
+DOM_GRAPHICS_CAPABILITIES="$(virsh domcapabilities | awk "/<graphics supported='yes'/{flag=1; next}/<\/graphics/{flag=0}flag")"
```
Comment (martinpitt) [inline pkg/machines/scripts/create_machine.sh]: This is indeed a much more elegant way than my `qemu  --qmp` query, nice!

The awk command could do with some spaces for readability:

    /<graphics supported='yes'/ {flag=1; next};  /<\/graphics/ {flag=0}; flag"

File: pkg/machines/scripts/create_machine.sh [hunk 2]
```diff
@@ -41,10 +26,17 @@ else
...
+GRAPHICS_PARAM=""
+if echo "$DOM_GRAPHICS_CAPABILITIES" | grep -q spice; then
+    GRAPHICS_PARAM="--graphics vnc,listen=127.0.0.1"
+fi
+if echo "$DOM_GRAPHICS_CAPABILITIES" | grep -q vnc; then
     GRAPHICS_PARAM="--graphics spice,listen=127.0.0.1 $GRAPHICS_PARAM"
```
Comment (martinpitt) [inline pkg/machines/scripts/create_machine.sh]: This is the wrong way around; you first grep for spice and then configure vnc, then vice versa.

**Comment by KKoukiou** [comment]
@martinpitt please don't merge this until @Antique gives ACK, because he mentioned that there might be libvirt bug for spice support detection.

**Comment by Antique** [comment]
Hi, so I new that there is something wrong with this detection but was not sure.  Now I've checked it and here is the result:

- it is exported in domcapabilities since libvirt-1.3.5, that should be safe
- for Spice it works perfectly, there is no issue
- for VNC it was recently fixed in QEMU-3.0.0, for older QEMU versions libvirt is not able to detect if VNC is disabled

The question is do we want to somehow workaround the VNC issue? I guess not, with upstream versions it's already fixed and the old code assumed that VNC is available anyway.

**Comment by martinpitt** [comment]
> for older QEMU versions libvirt is not able libvirt is not able to detect if VNC is disabled

That would mean that domcapabilities would always say that "vnc" is present? Then it seems safe (as the current master code just assumed that VNC was available). If it would say "always absent", then this would be a regression.

**Comment by KKoukiou** [comment]
> > for older QEMU versions libvirt is not able libvirt is not able to detect if VNC is disabled
> 
> That would mean that domcapabilities would always say that "vnc" is present? Then it seems safe (as the current master code just assumed that VNC was available). If it would say "always absent", then this would be a regression.

Exactly. So this is ready to land.

**Pull request merged**